src/google/protobuf/compiler/php/php_generator.cc

```diff
@@ -575,6 +575,9 @@ void GenerateFieldAccessor(const FieldDescriptor* field, const Options& options,
                          " is deprecated.', E_USER_DEPRECATED);\n        ")
           : "";
 
+  std::string deprecation_trigger_getter =
+      "if ($this->^name^ !== null) {\n    ^deprecation_trigger^\n}";
+
```

Comment:
This looks good, but I'd just update `deprecation_trigger` to include the conditional, rather than create a separate variable